### PR TITLE
Building react-based custom visual

### PR DESCRIPTION
**Pull request by blackleaden**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 source 'https://rubygems.org'
 
 group :development, :test do
-  gem 'jekyll', '~> 3.8.5'
+  gem 'jekyll', '~> 3.7.3'
```
Comment (uve) [inline Gemfile]: Why do you need yo downgrade jekyll version?

File: tutorials/building-react-based-custom-visual/getting-started.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+__3.__ Start developer server for your custom visual.
+  ```
+  pbiviz start
+  ```
+
+![updatesVisual](../images/updatesVisual.png)
+
+This basic visual represent updates count. Let's transform it to circle card, witch will represent some singular measure and it's title using React JS.
+
+---------
```
Comment (uve) [inline tutorials/building-react-based-custom-visual/getting-started.md]: add a link to the next step

File: tutorials/building-react-based-custom-visual/getting-viewport-properties.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+__5.__ Replace `width` and `height` rules in `style/visual.less`
+
+  ```css
+      min-width: 200px;
+      min-height: 200px;
+  ```
+
+  Now you can resize the viewport and the circle diameter will be equal to minimal size (width or height).
+
+---------
```
Comment (uve) [inline tutorials/building-react-based-custom-visual/getting-viewport-properties.md]: add a link to the next step

File: tutorials/building-react-based-custom-visual/receiving-props-from-powerbi-visual.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+              </div>
+          )
+      }
+  }
+  ```
+
+__2.__ Add some style for new markup by editing `styles/visual.less`.
+
+  ```css
+  .circleCard{
```
Comment (uve) [inline tutorials/building-react-based-custom-visual/receiving-props-from-powerbi-visual.md]: .circleCard{ -> .circleCard {

File: tutorials/building-react-based-custom-visual/receiving-props-from-powerbi-visual.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+          if(options.dataViews && options.dataViews[0]){
+              const dataView: DataView = options.dataViews[0];
+
+              ReactCircleCard.update({
+                  textLabel: dataView.metadata.columns[0].displayName,
+                  textValue: dataView.single.value.toString()
+              });
+          }
+      }
```
Comment (uve) [inline tutorials/building-react-based-custom-visual/receiving-props-from-powerbi-visual.md]: and Clear method for incorrect data

File: tutorials/building-react-based-custom-visual/receiving-props-from-powerbi-visual.md [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+      public componentWillUnmount() {
+          ReactCircleCard.updateCallback = null;
+      }
+  ```
+
+__5.__ Now you can test the component. Make sure that `pbiviz start` is run and all files are saved and update developing visual.
+
+![circleCard](../images/circleCard.png)
+
+---------
```
Comment (uve) [inline tutorials/building-react-based-custom-visual/receiving-props-from-powerbi-visual.md]: add a link to the next step

File: tutorials/building-react-based-custom-visual/receiving-props-from-powerbi-visual.md (hunk 5 above)
Comment (blackleaden) [reply tutorials/building-react-based-custom-visual/receiving-props-from-powerbi-visual.md]: Not needed here. Every kind of data available; the visual will be removed if data is removed.

**Pull request merged**